### PR TITLE
refactor(Hero): sync styles to latest DS

### PR DESCRIPTION
**Pull request by TylerAPfledderer**
Updates the Hero components so they match the latest DS changes.

Depends on https://github.com/ethereum/ethereum-org-website/pull/12853 to complete the sync

## Summary by CodeRabbit

- **New Features**
	- Enhanced internationalization support in the `ContentHero` component for better translation handling.
	- Improved responsive layout for the `ContentHero` and `HomeHero` components to adapt better on different screen sizes.
	- Refined layout and button arrangement in the `HubHero` component for improved user interaction.

- **Refactor**
	- Updated component structures in `ContentHero`, `HomeHero`, and `HubHero` for enhanced responsiveness and maintainability.

- **Style**
	- Adjusted visual properties such as height and spacing in the `HomeHero` component to improve visual appeal on various devices.

### CODE REVIEW

File: src/components/Hero/ContentHero/ContentHero.stories.tsx [hunk 1]
```diff
@@ -25,9 +26,17 @@ const meta = {
...
 export const ContentHero: StoryObj = {
+  parameters: {
+    // Set asPath in mock router so the Breadcrums component can render
+    // the "home" text with correct translation
+    nextjs: {
+      router: {
+        asPath: "/en",
+      },
+    },
+  },
```
Comment (TylerAPfledderer) [inline src/components/Hero/ContentHero/ContentHero.stories.tsx:37]: A good to know here, as this is integrated into Storybook with NextJS.

https://storybook.js.org/blog/integrate-nextjs-and-storybook-automatically/#nextrouter

File: src/components/Hero/ContentHero/ContentHero.stories.tsx [hunk 2]
```diff
@@ -25,9 +26,17 @@ const meta = {
...
   render: () => {
-    // eslint-disable-next-line react-hooks/rules-of-hooks
-    const { t } = useTranslation("page-learn")
+    const { t } = getI18n()
```
Comment (TylerAPfledderer) [inline src/components/Hero/ContentHero/ContentHero.stories.tsx:39]: Using this method is better here, so there are no errors/warnings of a hook being used outside of a component.

**Comment by minimalsm** [comment]
@coderabbitai review

**Comment by TylerAPfledderer** [comment]
- [ ] @TylerAPfledderer needs to review this thread for additional changes.

https://www.figma.com/file/L3EE4KGFU9hXqKITYH7uTI#788888481

**Comment by wackerow** [comment]
Thanks @TylerAPfledderer! Looks good at first glance, assigning to @nloureiro for design eyes

File: src/components/Hero/ContentHero/ContentHero.stories.tsx (hunk 1 above)
Comment (pettinarip) [reply src/components/Hero/ContentHero/ContentHero.stories.tsx:36]: Aha! interesting. Thanks

File: src/components/Hero/HomeHero/index.tsx [hunk 3]
```diff
@@ -33,11 +33,11 @@ const HomeHero = ({ heroImg }: HomeHeroProps) => {
           maxW="2xl"
         >
           <Morpher />
-          <VStack spacing="6">
+          <VStack spacing="6" maxW={{ lg: "2xl" }}>
             <Heading as="h1" size="2xl">
               {t("page-index:page-index-title")}
             </Heading>
-            <Text size="xl">{t("page-index:page-index-description")}</Text>
```
Comment (pettinarip) [inline src/components/Hero/HomeHero/index.tsx:40]: Is this correct? I see we are using `xl` in the DS.
image

File: src/components/Hero/HubHero/index.tsx [hunk 4]
```diff
@@ -70,12 +70,16 @@ const HubHero = ({
           </Heading>
           <Text size="lg">{description}</Text>
         </Stack>
-        <HStack justify={{ md: "center", xl: "start" }} spacing="4">
+        <Stack
+          direction={{ base: "column", md: "row" }}
```
Comment (pettinarip) [inline src/components/Hero/HubHero/index.tsx:74]: I don't see the figma file stacking the buttons on smaller res, is it correct? or the figma file should be updated?
image

**Review by pettinarip — CHANGES_REQUESTED** [review]
Nice @TylerAPfledderer. Left a few questions but its looking good.

File: src/components/Hero/HubHero/index.tsx (hunk 4 above)
Comment (TylerAPfledderer) [reply src/components/Hero/HubHero/index.tsx:74]: @nloureiro shows it off to the side in that file.



Though this file is starting to get a little out of date.

Nuno, I can update the file if you'd like!

Comment (pettinarip) [reply src/components/Hero/HubHero/index.tsx:74]: Ah got it! I think the stacked version is better to handle long i18n words.

File: src/components/Hero/HomeHero/index.tsx (hunk 3 above)
Comment (TylerAPfledderer) [reply src/components/Hero/HomeHero/index.tsx:40]: Whoops! I needed to change the `size` instead of removing it.

This has been fixed.

**Comment by wackerow** [comment]
Looking great! I see one conflict, and some resolved conversations after the last review... @TylerAPfledderer If you get a chance to clear the conflicts, and @pettinarip if things look good I'd say go ahead and pull in

**Review by pettinarip — APPROVED** [review]
Nice job @TylerAPfledderer

**Pull request merged**